### PR TITLE
Add gihub actions and update scalafmt

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

**Comment by tgodzik** [comment]
Ok, it seems to have run correctly on https://github.com/tgodzik/scaladex/actions/runs/356184937

I think it will not work locally until we merge it.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: CI
+on:
+  push:
+    branches:
+      - master 
+  pull_request:
```
Comment (adpi2) [inline .github/workflows/ci.yml:6]: We need to run the CI on pull requests that target `master` as well.

```
 pull_request:
   branches: 
      -  master
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          java-version: "adopt@1.8"
+      - uses: coursier/cache-action@v5
+      - name: Run tests
+        run: |
+          cd scaladex
+          ./bin/scalafmt --test
+          sbt "data/run elastic"
+          sbt test
+          sbt server/universal:packageBin
```
Comment (adpi2) [inline .github/workflows/ci.yml:35]: Could you make individual steps out of those?
So that it is easier to spot what is failing

**Review by adpi2 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tgodzik) [reply .github/workflows/ci.yml:6]: I think it's the default, but I will add it just in case

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tgodzik) [reply .github/workflows/ci.yml:35]: Sure!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tgodzik) [reply .github/workflows/ci.yml:6]: Added!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tgodzik) [reply .github/workflows/ci.yml]: Changed to separate steps:
https://github.com/tgodzik/scaladex/runs/1380895668?check_suite_focus=true

what do you think?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (adpi2) [reply .github/workflows/ci.yml:6]: okay I thought it was missing and that was why the CI was not running

File: .github/workflows/ci.yml (hunk 2 above)
Comment (adpi2) [reply .github/workflows/ci.yml]: Perfect thank you!

**Review by adpi2 — APPROVED** [review]

**Pull request merged**